### PR TITLE
[968] - Document Secure View with Collabora Online

### LINKED ISSUE #968: Document Secure View with Collabora Online
## What Needs to be Documented?

New oC Enterprise feature: Secure View (https://github.com/owncloud/enterprise/issues/2917)

## Where Does This Need To Be Documented?

- [ ] User docs
  - What is secure view?
  - How to use secure view if available (e.g., permissions in sharing)
- [ ] Admin docs
  - What is secure view?
  - Setup: oC Enterprise >= 10.2 / richdocuments app (currently unreleased, need to check about min. version) / Collabora Online >= 4.0.3 for good performance.
  - How to enable Secure View => Richdocuments admin settings
  - How to change the watermark wording
  - Setting defaults for permissions (secure view permissions + hint on sharing permission defaults)

## Why Should This Change Be Made? (Optional)

New Enterprise feature release with Server 10.2.

## What Type Of Content Change Is This? (Optional)
- [x] New Content Addition
- [ ] Old Content Deprecation
- [ ] Existing Content Simplification
- [ ] Bug Fix to Existing Content

## Which Manual Does This Relate To? (Optional)
- [x] Admin Manual
- [ ] Developer Manual
- [x] User Manual
- [ ] Android
- [ ] iOS
- [ ] Branded Clients
- [ ] Desktop Client
- [ ] Other

@PVince81 @mrow4a @settermjd

### PR DESCRIPTION
**Pull request by settermjd**
This PR extends the documentation to cover the new Secure View functionality in the user and admin manuals; including supporting images, and navigation file changes. 

It fixes #968.

### CODE REVIEW

**Comment by pmaier1** [comment]
Nice, thanks! Some feedback/additions:

- The UI you used is not the final one => final will look and behave like this: https://github.com/owncloud/enterprise/issues/2917#issuecomment-492712169
  - changes also apply to the default values in the settings
- Having Collabora Online set up and integrated is a prerequisite. The additional permissions are only shown if Collabora Online is properly configured and Secure View is enabled.
- It is important to emphasize that the documents do never leave the server when shared with Secure View. Collabora Online Server opens them and streams the tiles to the user's browser (like a video stream). Technically there's no way to extract the original document from the browser.
- Secure View is only available for certain file formats: docx, xlsx, pptx, odt, odp, ods, pdf
- Secure View comprises no edit / no download / no copy / watermarking + optional print and export
- Secure View always includes watermarks. The "old" UI was a little confusing about this. Without "edit" permission you basically have three options
  - Regular read-only share (download allowed)
  - Secure View without print/export
  - Secure View with print/export
- Users can share folders and single files with Secure View permissions
  - if a folder shared with Secure View contains unsupported files (e.g. JPG), they will just not be accessible
- Administrators can configure the watermark text in the settings

**Comment by settermjd** [comment]
Thanks for the excellent feedback. Is there a working installation that I can use to view the current version? I based the documentation in this PR off of the video that you shared.

File: modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc [hunk 1]
```diff
@@ -0,0 +1,25 @@
+= Secure View
+
+Secure View is a mode where users can only read, and optionally comment on, files.
```
Comment (pmaier1) [inline modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc]: - Commenting is not an option right now.
- You can optionally allow printing and exporting such files to PDF with watermarks included.

**Comment by pmaier1** [comment]
> Is there a working installation that I can use to view the current version? 

Unfortunately not as the changes are still in dev.

**Comment by settermjd** [comment]
@pmaier1, with respect to: "if a folder shared with Secure View contains unsupported files (e.g. JPG), they will just not be accessible", will the file(s) not be visible to the receiving user, or will the be visible, but not able to be viewed?

**Comment by settermjd** [comment]
@pmaier1, I've made further changes based on your feedback. Pls review and confirm if the content is complete, or needs further changes.

**Comment by pmaier1** [comment]
> @pmaier1, with respect to: "if a folder shared with Secure View contains unsupported files (e.g. JPG), they will just not be accessible", will the file(s) not be visible to the receiving user, or will the be visible, but not able to be viewed?

They will be visible but not accessible in any way.

File: modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+== How to Enable Secure View
+
+To enable _Secure View_, navigate to menu:Settings[Admin > Additional (Admin) > Collabora Online].
+At the bottom of the Collabora Online section, enable btn:[Enable Secure View].
+
+Once enabled, default share permissions for all users can, optionally, be enabled. 
+Currently, these default share permissions are:
+
+* *can view with watermark*.
```
Comment (pmaier1) [inline modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc]: `can view with watermark` => `Secure View (with watermarks)`
`can print/export` => `can print / export PDF`

> Documents can be printed or exported through Collabora Online. If this option is disabled, documents can only be viewed with a watermark.

^ That's not correct. If the option is enabled, you share in Secure View mode, meaning all the properties above apply at once (no edit, no dl, no print/export, no copy, watermarks). If you disable that option can "can edit" is also disabled you have a regular "read-only" share.

> Documents can be viewed through Collabora Online, but will be watermarked.

`can print / export PDF` is an addition to `Secure View (with watermarks)`. It additionally allows recipients to physically print documents or export them to PDF including the watermarks from within Collabora Online.

Generally, please note that in this section only default values for user/group sharing are set, meaning admins can specify that all shares are "Secure View" shares by default and the user intentionally has to change that or vice versa.

File: modules/user_manual/pages/online_collaboration.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+== Collabora Online
+
+{collabora-online-url}[Collabora Online] is a powerful {libreoffice-url}[LibreOffice]-based online office that supports all major document, spreadsheet and presentation file formats, and is integrable with ownCloud. 
+
+=== Secure View
+
+Secure View provides the ability to restrict access to documents, in _read-only_ shares, so that users can:
+
+. View a file with or without a custom watermark.
```
Comment (pmaier1) [inline modules/user_manual/pages/online_collaboration.adoc]: As above, "Secure View" always includes watermarks and the other properties at once. Print/export is an addition to that.

File: modules/user_manual/pages/online_collaboration.adoc [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+==== How to Use Secure View
+
+image::enterprise/collaboration/secure-view/collabora-online-administration.png[Configure Secure View in ownCloud Server.]
+
+To enable Secure View for an existing, or new, share:
+
+* Disable btn:[can edit]; Users will only be able to view documents.
+* Enable one or both of:
+** btn:[protect with watermarks]: Watermarked documents can be viewed through Collabora Online.
```
Comment (pmaier1) [inline modules/user_manual/pages/online_collaboration.adoc]: Needs to be adapted to the admin doc as well

**Comment by settermjd** [comment]
@pmaier1, further updates added, following your latest feedback.

File: modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+Secure View is a mode where users can place limitations on files and folders that are shared. 
+These limitations include:
+[[secure-view-mode-limitations]]
+* No copying
+* No downloading
+* No editing
+* Watermarking with optional print and export
++
+NOTE: Administrators can configure the watermark text.
+* Printing and exporting to PDF with watermarks included
```
Comment (pmaier1) [inline modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc]: Should be merged with the last point above, I guess?!

File: modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+== How to Enable Secure View
+
+To enable _Secure View_, navigate to menu:Settings[Admin > Additional (Admin) > Collabora Online].
+At the bottom of the Collabora Online section, enable btn:[Enable Secure View].
+
+Once enabled, default share permissions for all users can, optionally, be enabled. 
+Currently, these default share permissions are:
+
+* *Secure View (with watermarks)*. 
+   When enabled, files are shared in Secure View mode. In this mode , all xref:secure-view-mode-limitations[the Secure View limitations] are in-effect.
```
Comment (pmaier1) [inline modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc]: To add: `Secure View (with watermarks)` is mutually exclusive with `can edit`. Enabling one of them will disable the other.

File: modules/user_manual/pages/online_collaboration.adoc [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+NOTE: This feature is *not* available in the community edition.
+
+==== How to Use Secure View
+
+image::enterprise/collaboration/secure-view/collabora-online-administration.png[Configure Secure View in ownCloud Server.]
+
+To enable Secure View for an existing, or new, share:
+
+* Disable btn:[can edit]; Users will only be able to view documents.
```
Comment (pmaier1) [inline modules/user_manual/pages/online_collaboration.adoc]: To add: `Secure View (with watermarks)` is mutually exclusive with `can edit`. Enabling one of them will disable the other.

File: modules/user_manual/pages/online_collaboration.adoc (hunk 4 above)
Comment (pmaier1) [reply modules/user_manual/pages/online_collaboration.adoc]: This still needs to be adapted. There are only two new checkboxes as in this screenshot: https://github.com/owncloud/enterprise/issues/2917#issuecomment-492712169
For the user those are visible in the sharing dialog. For the admin they are visible in the Collabora Online settings to be able to set the default values.

File: modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc (hunk 5 above)
Comment (settermjd) [reply modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc]: Do you mean with the list of four limitations? It's merged, in a way, in that the admonition is part of the previous bullet point, as you can see in the screenshot below.

admonition-in-list

Comment (pmaier1) [reply modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc]: "Watermarking with optional print and export" and "Printing and exporting to PDF with watermarks included" is exactly the same :) would just remove "with optional print and export" from the bullet point before the note.

File: modules/user_manual/pages/online_collaboration.adoc (hunk 4 above)
Comment (settermjd) [reply modules/user_manual/pages/online_collaboration.adoc]: Is this still accurate for the admin view?

Comment (pmaier1) [reply modules/user_manual/pages/online_collaboration.adoc]: No. As mentioned, the two permission checkboxes have a different wording.

Comment (settermjd) [reply modules/user_manual/pages/online_collaboration.adoc]: I've updated the user docs. Just need confirmation on the admin doc screenshot, above.

File: modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc (hunk 5 above)
Comment (settermjd) [reply modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc]: Just applied your feedback.

Comment (pmaier1) [reply modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc]: Please add a last bullet point: "Optional printing and exporting to PDF with watermarks included"

File: modules/user_manual/pages/online_collaboration.adoc [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+To enable Secure View for an existing, or new, share:
+
+* Enable one (or both) of:
+** btn:[Secure View (with watermarks)]
+** btn:[can print / export PDF]: Documents can be printed and exported through Collabora Online.
+
+TIP: btn:[Secure View (with watermarks)] is mutually exclusive with btn:[can edit]. 
+Enabling one of them will disable the other.
+
+When "_protect with watermarks_" is enabled, any attempts to download the file will be blocked, as exemplified in the screenshot below.
```
Comment (pmaier1) [inline modules/user_manual/pages/online_collaboration.adoc]: "protect with watermarks" => "Secure View (with watermarks)"

File: modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc (hunk 5 above)
Comment (settermjd) [reply modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc]: Okay. Misunderstood previously.

File: modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+--
+NOTE: This option is only visible if btn:[Secure View (with watermarks)] is enabled.
+* Optional printing and exporting to PDF with watermarks included.
+
+When enabled, this mode allows documents to be can be printed or exported to PDF format—with a watermark—through Collabora Online.
+--
+
+NOTE: Admins can specify that all shares are "_Secure View_" by default and that the user has to intentionally change this setting, and vice versa,.
+
+When "_can view with watermark_" is enabled, any attempts to download the file will be blocked, as exemplified in the screenshot below.
```
Comment (pmaier1) [inline modules/admin_manual/pages/enterprise/collaboration/collabora_online_integration.adoc]: "can view with watermark" => "Secure View (with watermarks)"

**Comment by settermjd** [comment]
@pmaier1, thanks for your patience during the progression of this section of the documentation.

**Comment by settermjd** [comment]
@pmaier1, as per https://github.com/owncloud/enterprise/issues/2917#issue-370173710, is this documentation only relevant to 10.2?

**Comment by pmaier1** [comment]
> @pmaier1, as per owncloud/enterprise#2917 (comment), is this documentation only relevant to 10.2?

Well, below 10.2 you can't use it. So, yeah, probably only backport to 10.2 and later.

**Comment by voroyam** [comment]
Nicely done!

**Comment by voroyam** [comment]
@settermjd how about documenting the probably most important prerequisite? The App version that has secure view in it?

It's partially my bad for not noticing it. But if you would have tested it you could not have missed it.

https://marketplace.owncloud.com/apps/richdocuments

richdocuments app or how it's called in marketplace collabora online hast to be 2.2.0.

Please add it to the documentation in the prerequisite section.

image

**Pull request merged**